### ISSUE TITLE
Support aarch64-unknown-linux-musl

### ISSUE THREAD

**Issue by jiayihu** [deislabs/krustlet#466]
Not a request, I'm opening an issue to help anyone having the same issue in future. I've able to run krustlet with k3s on my raspberry pi 3B+ cluster without issues when I was using Ubuntu Server.

However, I decided to migrate to Alpine Linux to use less resources, but Alpine uses `musl` instead of `glibc`. This resulted in the arm64 binary not being runnable because it's compiled to be dynamically linked with `/lib/ld-linux-aarch64.so.1`. I had no luck with the compat packages at https://pkgs.alpinelinux.org/contents?file=ld-linux-*&path=&name=&branch=&repo=main&arch=

So I decided to build krustlet myself. To do so, you must wait for https://github.com/bytecodealliance/wasmtime/issues/2133 to be solved or patch `libc` yourself, like I did with the help of @lu-zero. Of course I assume one wants `krustlet-wasi`, `krustlet-wascc` had zero issues.

Then you cross-compile krustlet with `aarch64-unknown-linux-musl` target. On MacOS I used [cross](https://github.com/rust-embedded/cross) along with a custom image because they removed openssl support a while ago https://github.com/rust-embedded/cross/issues/229. You can find the Dockerfile I've used at https://github.com/jiayihu/fedra-thesis/blob/master/packages/fedra-compute/docker/Dockerfile. Or you can just use the image I've pushed configuring `Cross.toml`:

```toml
[target.aarch64-unknown-linux-musl]
image = "jiayihu/krustlet-musl:latest"
```

I was also successful by compiling to current target `aarch64-unknown-linux-gnu` with static linking. I've tried only the debug build and with rust-tls, but there shouldn't be surprises for release and openssl. Maybe that's the easiest solution to be applied in this repo. The only change needed is the compiler flag `-C target-feature=+crt-static`:

```
[target.aarch64-unknown-linux-gnu]
linker = "aarch64-linux-gnu-gcc"
rustflags = [ "-C", "target-feature=+crt-static"]
```

I can open a PR for it, what do you think?

**Comment by thomastaylor312** [comment]
Just to clarify, you can get that additional build target working purely using the compiler flags for static linking? If that is the case, I don't think we'd have a problem with adding this in if you want to PR it!

**Comment by thomastaylor312** [comment]
Sorry for the delay by the way. Most of us were out on holiday

**Comment by jiayihu** [comment]
No problem, I guessed you most of you were happily on holiday 😄 
Btw, yup I think the flag would be enough but I've tested it only once. It's officially documented here https://doc.rust-lang.org/reference/linkage.html#static-and-dynamic-c-runtimes, although I think it has been added to `rustc` only some months ago.

I'm working on something else right now, I'll test the flag with the openssl build and open a PR asap (and also for the other issue about Google Artifacts).
